### PR TITLE
[WIP] Asp.Net integration

### PR DESCRIPTION
**Pull request by sdanyliv**
Added new package for native support linq2db in Asp.Net projects.

### Taks to do
- [ ] Make options immutable
- [ ] Caching provider information
- [ ] Tests

### CODE REVIEW

File: NuGet/linq2db.AspNet.nuspec [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+	<metadata>
+		<id>linq2db.AspNet</id>
```
Comment (genusP) [inline NuGet/linq2db.AspNet.nuspec]: This package add integration with AspNetCore not with AspNet. This name is confusing.

Comment (sdanyliv) [reply NuGet/linq2db.AspNet.nuspec]: Good catch, will change name.

**Comment by hahn-kev** [comment]
I originally worked on the code here, and I'd like to contribute, but I'm not sure how. I made a pull request the branch `asp_net_integration` #2153 hopefully this should solve the immutability issue.

**Comment by MaceWindu** [comment]
We should rebase all related work on v3 branch I think

**Comment by hahn-kev** [comment]
> We should rebase all related work on v3 branch I think

I'd be happy to do that, though I'm not sure how. I changed the base to the release3.0 branch, and I've rebased the `asp_net_integration` branch onto release3.0 locally. But when I try to push it get's rejected. Do we need to make a new branch and PR for this?

**Comment by MaceWindu** [comment]
@sdanyliv could you provide  some directions here, as I'm not aware of details of this feature

**Comment by MaceWindu** [comment]
Continued in #2182

**Pull request closed without merging**